playbooks/installation/custom_certificates.adoc

```diff
@@ -0,0 +1,172 @@
+
+IMPORTANT: In order to do the above, master cluster public hostname and master cluster hostname  must be different.  If they’re the same, the named certificates will fail and will require a re-install. See the link:/playbooks/installation/#dns[Designing for DNS] section of the Install Guide.
+
+The Advantage of this approach is that it can be more flexible, as it allows you to start off relying completely on the self-signed certificates generated by OpenShift, and add-on custom trusted certificates to individual components as needed.
+
+The downside to this approach is that the internal infrastructure certificates remain self-signed, which may be percieved as bad practice by some Security or PKI teams. In reality the risk here is minimal, as the only clients that are trusting these certificates would be other components within the Cluster, and all external users and systems would be presented with custom trusted certificates.
+
+To see how to configure OpenShift to use this strategy, you can skip down to the link:#option-2-configuring-openshift-to-use-component-specific-custom-certificates[Configuring OpenShift to use Component-specific Custom Certificates] section.
+
+== Configuration
+
+There are two options for configuring custom certificates in OpenShift.
+
+=== Option 1: Configuring OpenShift to use an Intermediate CA
+
+In order to use this option, you will need to provision an Intermediate Certificate Authority Certificate and Key, and provide it to the installer to use to generate all of the OpenShift core certificates.
+
+IMPORTANT: This MUST happen on first install, or when running redeploy-openshift-ca playbook after which redeploy-certificates playbook have to be executed.
+
+Place the CA cert and key files on a directory accessible to Ansible, and add the following to your Ansible inventory.
+
+----
+openshift_master_ca_certificate={'certfile': '/path/to/ca.crt', 'keyfile': '/path/to/ca.key'}
+#
+# NOTE: CA certificate will not be replaced with existing clusters.
+# This option may only be specified when creating a new cluster or
+# when redeploying cluster certificates with the redeploy-certificates
+# playbook.
+----
+
+When ready, you may run the Ansible config playbook like so.
+
+----
+ansible-playbook -i c1-ocp.myorg.com/hosts /usr/share/ansible/openshift-ansible/playbooks/byo/config.yml
+----
+
+=== Option 2: Configuring OpenShift to use Component-specific Custom Certificates
+
+Alternatively, you can allow OpenShift to use its own CA to generate the internal certificates (self-signed) with which all of the internal components of OpenShift will authenticate. Custom certificates may be created and configured to be used individually by User-facing components of openshift. Discussion of each component is below.
+
+INFO: One way to simplify this is to have one certificate with multiple SAN names (console.example.com, metrics.example.com, logging.example.com), and to use the same certificate for all components.
```

Comment:
Just for reference: 

https://docs.openshift.com/container-platform/3.4/architecture/core_concepts/routes.html#secured-routes